### PR TITLE
Add --exit-if-inactive option to armadactl watch.

### LINKED ISSUE #200: watch should probably exit if the job succeeds or (permanently) fails
When `aramdctl watch` is watching a job live and it finishes, or it's called on an already finished job, it never exits:

```
$ ./bin/armadactl watch job-set-1                  
Watching job set job-set-1
Oct 23 17:54:39 | Queued:   0, Leased:   0, Pending:   0, Running:   0, Succeeded:   0, Failed
:   0, Cancelled:   0 | event: *api.JobSubmittedEvent, job id: 01dqww38n2japttpybyntwjnfc    
Oct 23 17:54:39 | Queued:   1, Leased:   0, Pending:   0, Running:   0, Succeeded:   0, Failed
:   0, Cancelled:   0 | event: *api.JobQueuedEvent, job id: 01dqww38n2japttpybyntwjnfc       
Oct 23 17:54:41 | Queued:   0, Leased:   1, Pending:   0, Running:   0, Succeeded:   0, Failed
:   0, Cancelled:   0 | event: *api.JobLeasedEvent, job id: 01dqww38n2japttpybyntwjnfc       
Oct 23 17:54:41 | Queued:   0, Leased:   0, Pending:   1, Running:   0, Succeeded:   0, Failed
:   0, Cancelled:   0 | event: *api.JobPendingEvent, job id: 01dqww38n2japttpybyntwjnfc      
Oct 23 17:54:43 | Queued:   0, Leased:   0, Pending:   0, Running:   1, Succeeded:   0, Failed
:   0, Cancelled:   0 | event: *api.JobRunningEvent, job id: 01dqww38n2japttpybyntwjnfc      
Oct 23 17:55:44 | Queued:   0, Leased:   0, Pending:   0, Running:   0, Succeeded:   1, Failed
:   0, Cancelled:   0 | event: *api.JobSucceededEvent, job id: 01dqww38n2japttpybyntwjnfc  
<just sits there>
```

I would expect success to result in exit code 0, and (permanent) failure to result in a non-0 exit code.

### PR DESCRIPTION
**Pull request by itamarst**
Fixes #200.

Remaining work:

* [x] Actually manually test if the command-line option does what it says :)
* [x] Add an end-to-end test (this may obviate need for the above).

### CODE REVIEW

File: cmd/armadactl/cmd/watch.go [hunk 1]
```diff
@@ -53,6 +55,9 @@ var watchCmd = &cobra.Command{
 					summary += fmt.Sprintf(" | event: %s, job id: %s", reflect.TypeOf(e), e.GetJobId())
 					log.Info(summary)
 				}
+				if exit_on_inactive && !state.HasActiveJobs() {
+					os.Exit(0)
```
Comment (JamesMurkin) [inline cmd/armadactl/cmd/watch.go]: I think if you just returned true here, it'd do the same and you wouldn't have to exit out of the application

Comment (itamarst) [reply cmd/armadactl/cmd/watch.go]: Done.

**Comment by itamarst** [comment]
Should be ready for review now.

**Comment by JamesMurkin** [comment]
Hey @itamarst, apologies we have been really slow on this.

So overall this change looks good, so I am happy to approve it and get it merged :).

A couple of things we could do in the future with it:
 - I don't overly love using the binary in the Integration test. We could just abstract the function and call that in the test.
 - We could also use this functionality in other places, such as in the Integration test itself. It has a function called submitJobsAndWatch which does something very similar. However this way of doing it is much nicer.
   - Maybe we could add the functionality to watch itself (we may need to refactor that function and have "watchOptions" or something.

However like I said, this looks good for now, approving.

**Pull request merged**